src/utils/promises.ts

```diff
@@ -0,0 +1,32 @@
+import { Context } from '../context';
+import { UnreachableError } from '../errors';
+
+export const DEFAULT_RETRIES = 2;
+
+export function asyncRetries<A extends unknown[], T>(
+	ctx: Context,
+	f: (...args: A) => Promise<T>,
+	retries = DEFAULT_RETRIES
+): (...args: A) => Promise<T> {
+	return async function (...args: A): Promise<T> {
+		for (let i = 0; i < retries; i += 1) {
+			try {
+				const result = await f(...args);
+				ctx.metrics.asyncRetriesTotal.inc({
+					retries: i,
+					success: true.toString(),
+				});
+				return result;
+			} catch (e: unknown) {
+				if (i === retries - 1) {
+					ctx.metrics.asyncRetriesTotal.inc({
+						retries: i,
+						success: false.toString(),
```

Comment:
```suggestion
						success: 'false',
```

Reply:
I considered both. Really no preference between the two, except I wanted to avoid ad-hoc strings